pkg/codegen/dotnet/gen.go

```diff
@@ -1609,7 +1630,7 @@ func (mod *modContext) genFunctionOutputVersion(w io.Writer, fun *schema.Functio
 			argumentName := LowerCamelCase(prop.Name)
 			propertyType := &schema.InputType{ElementType: prop.Type}
 			argumentType := mod.typeString(propertyType, "", true /* input */, false, true)
-			if prop.IsRequired() {
+			if prop.IsRequired() || outputOptions {
```

Comment:
Hmm, this is an interesting case. It means all optional args are no longer optional for the `InvokeOutputOptions` variant. Not sure it's worth it. I don't think we have any providers that are using `MultiArgumentInputs` yet and it's unclear to me if we will (I think we were originally looking at it for pulumi-std). I almost wonder if we should punt on including the InvokeOutputOptions variant for `MultiArgumentInputs` for now, adding a comment in the code explaining why. If someone really needs the dependsOn behavior, they can still pass `InvokeOutputOptions` to the existing function that accepts `InvokeOptions`.

Reply:
That sounds good to me, I added the comment here https://github.com/pulumi/pulumi/pull/17890/files#diff-c44352515fd31f42be144a763ab5771fe41195a15c2c1213d41a8cbc94dd69d4R1525-R1532